test/config-luatest/gh_10636_json_schema_test.lua

```diff
@@ -66,3 +68,37 @@ g.test_json_schema = function()
     }
     t.assert_equals(s:jsonschema(), expected)
 end
+
+g.test_json_schema_with_descriptions = function()
+    local app_section = cluster_config:jsonschema()['properties']['app']
```

Comment:
This test case should be either:

1. Placed in the first commit with some small testing schema (not config's one).
2. Placed in the third commit with using `config:jsonschema()` instead of `internal.config.cluster_config`.

This particular commit (the second one in the series) doesn't require a test, in my opinion, because it is tested in the next commit, where the public API for accessing the descriptions is added.